packages/api/src/modules/providers/subventia/validators/subventia.validator.ts

```diff
@@ -0,0 +1,89 @@
+import { ApplicationStatus } from "dto";
+import { isSiret } from "../../../../shared/Validators";
+import { ExcelDateToJSDate } from "../../../../shared/helpers/ParserHelper";
+
+export default class SubventiaValidator {
+    static sortDataByValidity(parsedData: object[]) {
+        const sortedData = parsedData.reduce(
+            (acc, currentLine) => {
+                if (this.isDataRowTypesValid(currentLine) && this.isDataRowCoherenceValid(currentLine)) {
+                    acc["valids"].push(currentLine);
+                } else {
+                    acc["invalids"].push(currentLine);
+                }
+                return acc;
+            },
+            { valids: [], invalids: [] },
+        );
+        return sortedData;
+    }
+
+    static validateDataRowTypes(parsedDataRow: object) {
+        if (!isSiret(parsedDataRow["SIRET - Demandeur"])) {
+            throw new Error(`INVALID SIRET FOR ${parsedDataRow["SIRET - Demandeur"]}`);
+        }
+
+        if (parsedDataRow["Date - Décision"]) {
+            try {
+                ExcelDateToJSDate(parseInt(parsedDataRow["Date - Décision"], 10));
```

Comment:
```suggestion
                ExcelDateToJSDate(parseInt(parsedDataRow["Date - Décision"]));
```
On pourrait le laisser mais je crois que 10 est la valeur par défaut de toute façon

Reply:
modifié